### ISSUE TITLE
SearchAfter method overload with params argument translated into wrong json request

### ISSUE THREAD

**Issue by bogdankolomiec706** [elastic/elasticsearch-net#5416]
**NEST/Elasticsearch.Net version**:
7.6.1

**Elasticsearch version**:
7.6.0

**Description of the problem including expected versus actual behavior**:

Usage of method overload `SearchAfter(params object[] searchAfter)` results in assignment of array of arrays to `search_after` field in request.

It looks like this
```
    "search_after": [
        [
            ... sorts ...
        ]
    ]
```
Intead of being like this
```
    "search_after": [
            ... sorts ...
    ],
```
As elastic API expects
https://www.elastic.co/guide/en/elasticsearch/reference/7.11/paginate-search-results.html#search-after

As result API returns this error
```
Elasticsearch.Net.ElasticsearchClientException: Request failed to execute. Call: Status code 400 from: POST /{index-name}/_search?typed_keys=true. ServerError: Type: parsing_exception Reason: "Expected [VALUE_STRING] or [VALUE_NUMBER] or [VALUE_BOOLEAN] or [VALUE_NULL] but found [START_ARRAY] inside search_after."
```

**Steps to reproduce**:
You can reproduce this case by passing any type except IList in order to trigger `SearchAfter(params object[] searchAfter)` overload.

Having 
```
class Doc
{
    public DateTime AddDate { get; set; }
}
```

This will use overload with `SearchAfter(IList<object> searchAfter)` and will generate correct request.
```
ISearchRequest SetupSearchRequest<T>(
    SearchDescriptor<T> descriptor, 
    IReadOnlyCollection<object> cursor) where T: Doc
{
    descriptor = descriptor.Sort(s => s.Ascending(f => f.AddDate)).Size(10);
    if (cursor != null)
    {
        descriptor = descriptor.SearchAfter(cursor.ToList()); //explicit cast to list
    }
    return descriptor;
}
```

This will use overload `SearchAfter(params object[] searchAfter)` and will generate incorrect request
```
ISearchRequest SetupSearchRequest<T>(
    SearchDescriptor<T> descriptor, 
    object cursor) where T: Doc
{
    descriptor = descriptor.Sort(s => s.Ascending(f => f.AddDate)).Size(10);
    if (cursor != null)
    {
        descriptor = descriptor.SearchAfter(cursor);
    }
    return descriptor;
}
```

**Comment by stevejgordon** [comment]
Hi @bogdankolomiec706. I don't believe this is a bug but simply how params functions. It can accept either, one or more objects, or an array of objects. Since it seems you're passing an `IReadOnlyCollection` it's likely that it's being treated as the first object for the object array, resulting in an array within an array in serialisation. If you call `cursor.ToArray()` it should be interpreted as you intend.

**Comment by bogdankolomiec706** [comment]
Hi @stevejgordon , IReadOnlyCollection is a type of Sorts property of IHit<out TDocument> interface, so I just passed back to library what it has returned to me from previous response.

**Comment by stevejgordon** [comment]
@bogdankolomiec706 That's a fair criticism of the current design which I'm going to address by adding an IEnumerable<object> overload for the fluent syntax. This will make using the sort from the previous search a cleaner experience. The params overload is likely to be more rarely needed but does have potential use cases. We may review this design further in a future major release but hopefully, my upcoming PR will improve consumption patterns such as yours.

**Comment by stevejgordon** [comment]
PR is merged for 7.12.0 release.
